### PR TITLE
Add auto-refresh on index page

### PR DESCRIPTION
**Pull request by tjayl**
Add an auto refresh to the index page.

### CODE REVIEW

**Comment by fgimian** [comment]
Hmm, I don't think we should do this for refreshing personally.  Instead, we really need to investigate an AJAX-based solution like Angular.js or similar.  I'll test this out tomorrow but I'm pretty set on using a strategy that involved Javascript to dynamically update elements on the page instead of refreshing the entire page via a meta tag.

**Comment by nibalizer** [comment]
I am personally happy with this solution until an AJAX solution emerges. @tjayl can you rebase this and we can merge it?

**Comment by nibalizer** [comment]
@tjayl rebase?

**Comment by tjayl** [comment]
Sorry dude,
I have put all my changes in 245e625

**Comment by bastelfreak** [comment]
@tjayl could you take a look again? it seems like a rebase is needed here.

**Comment by tjayl** [comment]
I am unsure what you are expecting of me here. I have put all of the changes for this into one commit.

If i do a `git rebase` i only get 'Current branch master is up to date.'

**Comment by bastelfreak** [comment]
You've to do a rebase against the master:
```
git checkout master
git pull # to update your master to the latest one
git checkout auto-refresh
git rebase -i master
```

this is untested and based on my memory, use the commands with caution. If you experience any issues you can ask here or also in #puppet-community on freenode

**Comment by tjayl** [comment]
Thanks bastelfreak.

I have done that and pushed it back to my repo.

**Comment by bastelfreak** [comment]
did you push it in the wrong branch? I can't see anything changed here :(

git is pure magic

**Comment by tjayl** [comment]
I pushed it into the master branch on my repo:

```
git remote -v
origin	https://github.com/tjayl/puppetboard.git (fetch)
origin	https://github.com/tjayl/puppetboard.git (push)
```

```
git branch -v
  auto-refresh 245e625 Add auto refresh to the index page
* master       245e625 Add auto refresh to the index page
```

**Comment by janorn** [comment]
You need to add the original repo as an upstream.

https://help.github.com/articles/configuring-a-remote-for-a-fork/

**Comment by janorn** [comment]
Or follow these steps.

https://help.github.com/articles/merging-an-upstream-repository-into-your-fork/

**Comment by tjayl** [comment]
Thanks for those guides.

I think i have managed to do it.

**Comment by bastelfreak** [comment]
After the rebase you've to push (probably push -f) into your auto-refresh branch, than the changes will appear here. I can't see anything changed here. If you want, we can fiddle that out if you join #puppet-community.

**Comment by tjayl** [comment]
Got it.

If there are still issues, i will jump on IRC to resolve them

**Comment by bastelfreak** [comment]
great, you based it \o/

but you missed two pieces off git diff code. could you rebase again and squash it down to one commit?

File: puppetboard/app.py [hunk 1]
```diff
@@ -192,8 +193,12 @@ def index(env):
         metrics=metrics,
         nodes=nodes_overview,
         stats=stats,
+<<<<<<< HEAD
+        refreshrate=refreshrate
+=======
         envs=envs,
         current_env=env
+>>>>>>> f63a0cefcb71dfcd309fa508837dc131f90862a4
```
Comment (bastelfreak) [inline puppetboard/app.py]: please remove this line

File: puppetboard/app.py [hunk 2]
```diff
@@ -192,8 +193,12 @@ def index(env):
         metrics=metrics,
         nodes=nodes_overview,
         stats=stats,
+<<<<<<< HEAD
```
Comment (bastelfreak) [inline puppetboard/app.py]: please remove this line

File: puppetboard/default_settings.py [hunk 3]
```diff
@@ -35,3 +36,4 @@
                     ('Architecture',   'hardwaremodel'     ),
                     ('Kernel Version', 'kernelrelease'     ),
                     ('Puppet Version', 'puppetversion'     ), ]
+>>>>>>> f63a0cefcb71dfcd309fa508837dc131f90862a4
```
Comment (bastelfreak) [inline puppetboard/default_settings.py]: please remove this line

File: puppetboard/app.py [hunk 4]
```diff
@@ -192,8 +193,12 @@ def index(env):
...
+        refreshrate=refreshrate
+=======
```
Comment (bastelfreak) [inline puppetboard/app.py]: please remove this line (this is a diff between old and new, be sure to only merge the needed params)

**Comment by tjayl** [comment]
rebase done

**Comment by bastelfreak** [comment]
the code looks fine, could you squash the two commits into one? Than we should finally be ready to merge \o/

**Comment by janorn** [comment]
https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/

**Comment by tjayl** [comment]
It doesn't seem to want to squash 245e625 and d44152f

I am guessing because 245e625 was committed over a year ago

**Comment by bastelfreak** [comment]
Would it be possible for you to create a new branch with just a single commit and a new PR? That could be easier than squashing this one.

**Comment by tjayl** [comment]
Done it (at last)

**Comment by bastelfreak** [comment]
thanks, looks perfect!
can somebody merge this or do the docs need to be extended?

**Pull request merged**